src/main/kotlin/biokotlin/util/GetVCFVariants.kt

```diff
@@ -0,0 +1,212 @@
+        val jobs = vcfReaders
+            .map { reader ->
+                CoroutineScope(Dispatchers.IO).launch {
+
+                    while (reader.variant() != null && reader.variant()!!.endPosition < position) {
+                        reader.advanceVariant()
+                    }
+
+                }
+
+            }
+
+        // Wait for all readers to advance to the first variant that
+        // is within the position range
+        jobs.forEach { it.join() }
+
+        return vcfReaders.associate { reader ->
+
+            var variant = reader.variant()
+            if (variant != null && variant.contains(position)) {
+                Pair(reader, variant)
+            } else {
+                Pair(reader, null)
+            }
+
+        }
+
+    }
+
+    suspend fun forAll(channel: Channel<List<Pair<Int, List<SimpleVariant?>>>>) = withContext(Dispatchers.IO) {
+
+        val stepSize = 10000
+
+        var lowestPosition: Position? = vcfReaders
+            .mapNotNull { it.variant()?.startPosition }
+            .minOrNull()
+
+        while (lowestPosition != null) {
+
+            val currentContig = lowestPosition.contig
+            var currentStart = lowestPosition.position
+
+            var rangeMaps: Array<RangeMap<Int, SimpleVariant>?> = Array(vcfReaders.size) { null }
+
+            do {
+
+                val nextPositions = vcfReaders.mapIndexed { index, reader ->
+                    val thisStart = currentStart
+                    val thisRangeMap = rangeMaps[index]
+                    val thisReader = reader
+                    async {
+                        nextPositions(
+                            thisStart,
+                            Position(currentContig, thisStart + stepSize - 1),
+                            thisRangeMap,
+                            thisReader
+                        )
+                    }
+                }
+
+                rangeMaps = Array(vcfReaders.size) { null }
+                val startPositions = mutableSetOf<Int>()
+                for ((index, deferred) in nextPositions.withIndex()) {
+                    val nextPositionsResult = deferred.await()
+                    rangeMaps[index] = nextPositionsResult.rangeMap
+                    startPositions.addAll(nextPositionsResult.positions)
+                }
+
+                val thisRangeMaps = rangeMaps
+                val thisStartPositions = startPositions.toSet()
+                channel.send(
+                    processBlock(
+                        thisStartPositions,
+                        vcfReaders.size,
+                        thisRangeMaps
+                    )
+                )
+
+                currentStart += stepSize
+
+            } while (!vcfReaders.all { it.variant() == null } &&
+                vcfReaders.mapNotNull { it.variant()?.startPosition }
+                    .find { it.contig == currentContig } != null)
+
+            // TODO - Finding lowest based on contig / position
```

Comment:
The code is finding the lowest based on contig/position, so can remove the "TODO" ?